### ISSUE TITLE
Comments on 'Writing Data'

### ISSUE THREAD

**Issue by katieannemills** [katieannemills/blog#10]
Let me know your thoughts on my [Writing Data](http://billmills.github.io/blog/writing-data) post here.

**Comment by khinsen** [comment]
A really nice overview!

I'd add a few sentences about XML to the "hiearchical data" section. Depending on which field of science you are in, XML can be much more important than JSON, so it deserves to be mentioned if only to point out that it's basically the same idea.

I also miss the good general advice to use community-standard formats as much as possible but not more. In other words, if you can't use an existing format without torturing it (very good reminder at the end of your post), then better make up your own.

**Comment by katieannemills** [comment]
@khinsen thanks for the comment! re: tortuous standards, I think it depends on the torture :) - if there's something really pathological about the standard, then you're right - time to leave the mistakes of the past in the past. But, sometimes excruciating data formats (see character stanzas) can be obfuscated behind good tooling; taking the time to write a nice, stand alone packer/unpacker is (a) something you'll mostly have to do anyway to consume the data you got from your colleagues, and (b) something that might help the standard be less excruciating in future. All that said, I only cling so fiercely to community standards because it is _so darn hard_ to get people to agree standards in the first place; if that mountain of diplomacy has already been scaled, it might be worth a bit of pain to keep the kingdom from fragmenting again.

**Comment by khinsen** [comment]
@BillMills I fully agree with the intention of maintaining good community standard. What I meant by "torture" is the kind of misuse that actually causes technical fragmentation, although superficially a single standard is maintained. My favorite example is the [PDB file format](http://www.wwpdb.org/documentation/file-format). It looks like a well-defined standard, and plenty of programs claim to support it, but interoperability is low except for the most basic use cases. One reason for the fragmentation is that the old format from the 1970s was becoming insufficient, so people worked around its limitations. But the main reason was that few software developers even cared to look at its definition. People just took a few examples and wrote a subroutine that produced something that looked similar.

The PDB format is also an interesting case study because of its unintended longevity. Its inventor, the PDB designed and published a replacement (mmCIF) more than 15 years ago, because the limitations already were a problem. But the definitive transition to the new format was announced only last year. The reason was an enormous resistance to the change, because of the many programs that needed fixing, many of them being no longer actively maintained.
